### PR TITLE
fix: Changed fragments to activities for evey sensors

### LINKED ISSUE #969: Unhandled threads in sensors fragment
**Actual Behaviour**

When any sensor is opened from the given list, then on pressing back button crashes the app

**Expected Behaviour**

The app shouldn't crash, rather main fragment main fragment of sensor activity should pop up

**Steps to reproduce it**

Go to sensors and use any sensor from the list and then press back key

**LogCat for the issue**

Not available

**Screenshots of the issue**

**Would you like to work on the issue?**

Yes

NOTE: The toast message only shows the number of running threads while app crashed

### PR DESCRIPTION
**Pull request by harsh-2711**
Fixes #969 #970 #984 #986 

Changes: Changed fragments to activities for every sensor and removed the placeholder fragment from main sensor activity and updated the sensor activity to show the sensors list solving orientation issue. Also provided suitable margins to the sensor dock and the layout to provide better UI. Also the issue of orientation is handled by creating activities.

Screenshots for the change: 

APK for testing: [app-debug.zip](https://github.com/fossasia/pslab-android/files/2078495/app-debug.zip)

### CODE REVIEW

File: app/src/main/java/org/fossasia/pslab/activity/SensorActivity.java [hunk 1]
```diff
@@ -1,140 +1,183 @@
...
-                    play = false;
+                if (scienceLab.isConnected()) {
+                    new Thread(scanRunnable).start();
                 } else {
-                    playPauseButton.setImageResource(R.drawable.pause);
-                    play = true;
-                    if (!indefiniteSamplesCheckBox.isChecked()) {
-                        counter = Integer.parseInt(samplesEditBox.getText().toString());
-                    }
+                    Snackbar snackbar = Snackbar.make(coordinatorLayout, "Device not connected", Snackbar.LENGTH_SHORT);
```
Comment (CloudyPadmal) [inline app/src/main/java/org/fossasia/pslab/activity/SensorActivity.java]: Use resources

**Comment by harsh-2711** [comment]
@Vikum94 I think the present scenario is OK as showing list of sensors that are not available doesn't looks good. Also for how to connect purpose the user can refer to documentation that we will make as said by @mariobehling if the sensor isn't available in his/her device. And I think all the compatible sensors are shown in the list if the device is connected. This issue is solved under current PR only.

**Comment by harsh-2711** [comment]
@CloudyPadmal Please remove conflicts tag

File: app/src/main/java/org/fossasia/pslab/activity/SensorActivity.java [hunk 2]
```diff
@@ -1,140 +1,185 @@
...
-    TextView timeGapLabel;
-    @BindView(R.id.imageButton_play_pause_sensor)
-    public ImageButton playPauseButton;
-    public boolean play;
-    public boolean runIndefinitely;
-    public static int counter;
-    public int timeGap;
+    private I2C i2c;
+    private ScienceLab scienceLab;
+    private LinkedHashMap<Integer, String> sensorAddr = new LinkedHashMap<>();
```
[the remark this answers is not in the document]
Comment (harsh-2711) [reply app/src/main/java/org/fossasia/pslab/activity/SensorActivity.java]: To preserve the order of sensors everytime

**Comment by Vikum94** [comment]
@CloudyPadmal Will it be good to reformat manifest file? It's not formatted, but there will be merge conflicts in other PRs

**Comment by harsh-2711** [comment]
@Vikum94 I removed the changes in manifest file. Now it won't create any conflicts

File: app/src/main/java/org/fossasia/pslab/sensors/SensorADS1115.java [hunk 3]
```diff
@@ -29,47 +32,61 @@
...
         super.onCreate(savedInstanceState);
+        setContentView(R.layout.sensor_ads1115);
+
+        sensorDock = (RelativeLayout) findViewById(R.id.sensor_control_dock_layout);
+        indefiniteSamplesCheckBox = (CheckBox) findViewById(R.id.checkBox_samples_sensor);
+        samplesEditBox = (EditText) findViewById(R.id.editBox_samples_sensors);
+        timeGapSeekbar = (SeekBar) findViewById(R.id.seekBar_timegap_sensor);
+        timeGapLabel = (TextView) findViewById(R.id.tv_timegap_label);
+        playPauseButton = (ImageButton) findViewById(R.id.imageButton_play_pause_sensor);
+        setSensorDock();
```
Comment (Vikum94) [inline app/src/main/java/org/fossasia/pslab/sensors/SensorADS1115.java]: You can use butter knife to bind views

Comment (harsh-2711) [reply app/src/main/java/org/fossasia/pslab/sensors/SensorADS1115.java]: Sure..

**Comment by CloudyPadmal** [comment]
@harsh-2711 let's merge this as a work in progress for other devs to work on sensors taking this as a base. Please resolve this conflict!

**Comment by harsh-2711** [comment]
@CloudyPadmal Changes done :+1:

**Pull request merged**